### ISSUE TITLE
Performance reproduction: can this code reach the test acc(e.g. fig2(b) ) claimed in the paper? 

### ISSUE THREAD

**Issue by Youth-49** [anonymifish/fed-distribution-matching#1]
I wonder is it the official code implementation of the paper?

can this code achieve the performance claimed in the paper? The code set the same hyper-parameters as the paper and I'd like to know if you can reproduce the result in figure 2 in the paper. 

I run the run.sh and got the result of 61% test accuracy in cifar10 under Dir_0.5(10), but FedAvg can easily achieve 70% test accuracy with local epoch=10, lr=0.01, momentum=0.5. The test accuracy curve against the communication of FedDM and FedAvg round are as below.

![net](https://github.com/anonymifish/fed-distribution-matching/assets/59035302/323b52e6-78f4-46e5-8361-df139171eaf1)
![vanilla_fed_cifar10_ConvNet_20_C1_iidFalse_E10_nclient10_a0 5](https://github.com/anonymifish/fed-distribution-matching/assets/59035302/d4c06f9e-8078-46c6-8fa6-6d12b43f4e42)

**Comment by Ariatao** [comment]
Hello, I also encountered the same problem when reproducing this repository code, but I did not achieve such a fast convergence speed using FedAvg under Dir (0.5) partitioning (about 63% test accuracy at 20th round, CIFAR10, simple-cnn). I use implementations from [https://github.com/Xtra-Computing/NIID-Bench](https://github.com/Xtra-Computing/NIID-Bench ), all parameters keep default, which same as the baselines implemented in the paper FedDM. I am currently considering whether the decrease in accuracy was caused by reasons such as not performing DSA operations during the training process. If there is any progress,  we can discuss it together.

**Comment by aikedaerC** [comment]
I got results similar to `FedDM` using centralized training.

**Comment by xirihao** [comment]
> I got results similar to `FedDM` using centralized training.

Could you please give me some advice on how to achieve the effect in the paper?
